Sources/XMLCoder/Auxiliaries/XMLElement.swift

```diff
@@ -0,0 +1,280 @@
+    var children: [String: [_XMLElement]] = [:]
+    
+    internal init(key: String, value: String? = nil, attributes: [String: String] = [:], children: [String: [_XMLElement]] = [:]) {
+        self.key = key
+        self.value = value
+        self.attributes = attributes
+        self.children = children
+    }
+    
+    static func createRootElement(rootKey: String, object: ArrayBox) -> _XMLElement? {
+        let element = _XMLElement(key: rootKey)
+        
+        _XMLElement.createElement(parentElement: element, key: rootKey, object: object)
+        
+        return element
+    }
+    
+    static func createRootElement(rootKey: String, object: DictionaryBox) -> _XMLElement? {
+        let element = _XMLElement(key: rootKey)
+        
+        _XMLElement.modifyElement(element: element, parentElement: nil, key: nil, object: object)
+        
+        return element
+    }
+    
+    fileprivate static func modifyElement(element: _XMLElement, parentElement: _XMLElement?, key: String?, object: DictionaryBox) {
+        let attributesBox = object[_XMLElement.attributesKey]?.dictionary
+        element.attributes = attributesBox?.unbox().mapValues { $0.xmlString } ?? [:]
+        
+        let objects = object.filter { key, _value in key != _XMLElement.attributesKey }
+        
+        for (key, box) in objects {
+            switch box {
+            case .null(let box):
+                _XMLElement.createElement(parentElement: element, key: key, object: box)
+            case .bool(let box):
+                _XMLElement.createElement(parentElement: element, key: key, object: box)
+            case .decimal(let box):
+                _XMLElement.createElement(parentElement: element, key: key, object: box)
+            case .signedInteger(let box):
+                _XMLElement.createElement(parentElement: element, key: key, object: box)
+            case .unsignedInteger(let box):
+                _XMLElement.createElement(parentElement: element, key: key, object: box)
+            case .floatingPoint(let box):
+                _XMLElement.createElement(parentElement: element, key: key, object: box)
+            case .string(let box):
+                _XMLElement.createElement(parentElement: element, key: key, object: box)
+            case .array(let box):
+                _XMLElement.createElement(parentElement: element, key: key, object: box)
+            case .dictionary(let box):
+                _XMLElement.createElement(parentElement: element, key: key, object: box)
+            }
+        }
+        
+        if let parentElement = parentElement, let key = key {
+            parentElement.children[key] = (parentElement.children[key] ?? []) + [element]
+        }
+    }
+    
+    fileprivate static func createElement(parentElement: _XMLElement, key: String, object: NullBox) {
+        let element = _XMLElement(key: key)
+        parentElement.children[key] = (parentElement.children[key] ?? []) + [element]
```

Comment:
couldn't this be simplified as `parentElement.children[key]?.append(element)`?

Reply:
I think so, yes.

Reply:
Fixed.